amlb/framework_definitions.py

```diff
@@ -55,17 +55,12 @@ def _sanitize_and_add_defaults(frameworks, resource):
     _add_defaults_to_frameworks(frameworks, resource)
 
 
-def _add_and_normalize_names(frameworks: Namespace):
+def _add_framework_name(frameworks: Namespace):
     """ Converts each framework definition to lowercase and adds a 'name' field. """
     framework_names = dir(frameworks)
     for name in framework_names:
         framework = frameworks[name]
-        framework.name = name.lower()
-        if name.lower() != name:
-            del frameworks[name]
-            frameworks[name.lower()] = framework
-        if "extends" in framework:
-            framework.extends = framework.extends.lower()
+        framework.name = name
```

Comment:
(I can't make a github suggestion because it involves deleted and added lines...)
The docstring needs to be updated too, and the code can now be simplified (since we use the original name).

```python
def _add_framework_name(frameworks: Namespace):
    """ Adds a 'name' field to each framework. """
    for name, framework in frameworks:
        framework.name = name
```

At this point we might also consider removing the separate function call instead. 
But I suppose it is nice we maintain more isolated tests by keeping the function separate as is.